src/pybind/mgr/mgr_util.py

```diff
@@ -978,8 +943,27 @@ def wrapper(*args: Any, **kwargs: Any) -> T:
 def password_hash(password: Optional[str], salt_password: Optional[str] = None) -> Optional[str]:
     if not password:
         return None
+
     if not salt_password:
-        salt = bcrypt.gensalt()
-    else:
-        salt = salt_password.encode('utf8')
-    return bcrypt.hashpw(password.encode('utf8'), salt).decode('utf8')
+        salt_password = ''
+
+    __bcrypt_password = """import bcrypt
+
+password = "{}"
+salt_password = "{}"
+
+if not salt_password:
+    salt = bcrypt.gensalt()
+else:
+    salt = salt_password.encode('utf8')
+
+print(bcrypt.hashpw(password.encode('utf8'), salt).decode('utf8'))""".format(password, salt_password)
+
+    import subprocess
+    result = subprocess.run(["/usr/bin/python3", "-c", __bcrypt_password], capture_output=True, text=True)
```

Comment:
Oh, I think I missed this part... This behaves the same as an `eval()/exec()` call, and is worse because if fed with user supplied data, so this is basically introducing a code-injection vulnerability.

What happens, for example, if a user tries to log in with  the password `";import subprocess;subprocess.run("sudo rm -rf /var/lib/ceph");`?

Reply:
That bit of code was "cough" "cough" ... highly experimental. 
It was part of the initial framework I put together just to prove that it could be done. I'm very glad you spotted it.